ethstorage/p2p/protocol/syncclient.go

```diff
@@ -1260,6 +1267,40 @@ func (s *SyncClient) ReportPeerSummary() {
 	}
 }
 
+func (s *SyncClient) PurgeBadPeers(closePeerFunc func(peer.ID) error) {
+	ticker := time.NewTicker(time.Minute)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			peers := s.getBadPeers()
+			for _, p := range peers {
+				err := closePeerFunc(p)
+				if err != nil {
+					log.Info("Purge bad peer failed", "peer", p.String(), "error", err.Error())
+				}
+				log.Debug("Purge bad peer", "peer", p.String())
+			}
+		case <-s.resCtx.Done():
+			log.Info("P2P PurgeBadPeers stop")
+			return
+		}
+	}
+}
+
+// TODO we can add more rules for bad peers
+func (s *SyncClient) getBadPeers() []peer.ID {
```

Comment:
The PR description needs to be updated if the changes are more than just lowering the log level.